src/test/java/uk/gov/companieshouse/api/testdata/spec/CompanySpecTest.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.companieshouse.api.testdata.spec;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException;
+import jakarta.validation.ConstraintViolation;
+import jakarta.validation.Validation;
+import jakarta.validation.Validator;
+import jakarta.validation.ValidatorFactory;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.api.testdata.model.rest.CompanySpec;
+import uk.gov.companieshouse.api.testdata.model.rest.Jurisdiction;
+import java.util.Set;
+
+@ExtendWith(MockitoExtension.class)
+class CompanySpecTest {
+
+    // Test that a spec is not accepting field attributes that are not defined in the CompanySpec class
+    @Test
+    void testDeserializationFailsOnUnknownProperties() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        String invalidJson = "{ \"jurisdiction\": \"england-wales\", \"companystatus\": \"active\" }";
+        assertThrows(UnrecognizedPropertyException.class, () -> objectMapper.readValue(invalidJson, CompanySpec.class));
+    }
+
+    // Test that a spec is accepting field attributes that are defined in the CompanySpec class
+    @Test
+    void testDeserializationSucceedsWithValidProperties() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        String validJson = "{ \"jurisdiction\": \"england-wales\", \"company_status\": \"dissolved\" }";
+        assertDoesNotThrow(() -> {
```

Comment:
I don't think we need this assert. If it threw an exception, wouldn't the test fail anyway?

I'd assert the returned object contains the right values instead

Reply:
removed these tests as its not required